unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs

```diff
@@ -77,21 +107,57 @@ private void LoadAndShowOwnedWearables(UserProfile userProfile)
                                       var containedWearables = wearables.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT)
                                          .Where(wearable => wearableCatalogBridge.IsValidWearable(wearable.id));
                                       view.SetCollectibleWearables(containedWearables.ToArray());
+                                      view.SetCollectibleWearablesLoadingActive(false);
                                   })
                                  .Catch(Debug.LogError);
         }
 
         private void LoadAndShowOwnedEmotes(UserProfile userProfile)
         {
+            view.SetCollectibleEmotesLoadingActive(true);
             emotesCatalogService.RequestOwnedEmotes(userProfile.userId)
                                  .Then(emotes =>
                                   {
                                       WearableItem[] emoteItems = emotes.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT).ToArray();
                                       view.SetCollectibleEmotes(emoteItems);
+                                      view.SetCollectibleEmotesLoadingActive(false);
                                   })
                                  .Catch(Debug.LogError);
         }
 
+        private async UniTask LoadAndShowOwnedNamesAsync(UserProfile userProfile)
+        {
+            view.SetCollectibleNamesLoadingActive(true);
+            using var pagePointer = namesService.GetPaginationPointer(userProfile.userId, MAX_NFT_COUNT, CancellationToken.None);
+            var response = await pagePointer.GetPageAsync(1, CancellationToken.None);
+            var namesResult = Array.Empty<NamesResponse.NameEntry>();
+
+            if (response.success)
+                namesResult = response.response.Names.ToArray();
+            else
+                Debug.LogError("Error requesting names lambdas!");
+
+            view.SetCollectibleNames(namesResult);
+            view.SetCollectibleNamesLoadingActive(false);
+        }
+
+        private async UniTask LoadAndShowOwnedLandsAsync(UserProfile userProfile)
+        {
+            view.SetCollectibleLandsLoadingActive(true);
+            // TODO (Santi): Use userProfile.userId here!!
+            using var pagePointer = landsService.GetPaginationPointer(userProfile.userId, MAX_NFT_COUNT, CancellationToken.None);
+            var response = await pagePointer.GetPageAsync(1, CancellationToken.None);
+            var landsResult = Array.Empty<LandsResponse.LandEntry>();
+
+            if (response.success)
+                landsResult = response.response.Lands.ToArray();
+            else
+                Debug.LogError("Error requesting lands lambdas!");
+
+            view.SetCollectibleLands(landsResult);
+            view.SetCollectibleLandsLoadingActive(false);
+        }
+
         private async UniTask<string> FilterContentAsync(string filterContent) =>
```

Comment:
Please change it to accept `CancellationToken`, also please `Filter` to accept it as well and handle properly: it should not take long